target/arc/arconnect.c

```diff
@@ -205,6 +210,109 @@ static void arconnect_intercore_intr_unit_cmd(CPUARCState *env, enum arconnect_c
 #define ARCON_COMMAND(V) (V & 0xff)
 #define ARCON_PARAM(V) ((V >> 8) & 0xffff)
 
+static void arc_cirq_raise(CPUARCState *env, uint16_t cirq)
+{
+    unsigned int max_cpus = 4; /* TODO: make this variable */
+    enum idu_mode_enum mode = env->arconnect.idu_data[cirq].mode;
+    uint8_t counter = env->arconnect.idu_data[cirq].counter;
+    int dest = env->arconnect.idu_data[cirq].dest;
+    bool is_first_acknoledge = false;
+    uint8_t core_id = 0;
+
+    switch(mode) {
+    case ROUND_ROBIN:
+	/* Round robin */
+	do {
+	    core_id = (counter + cirq) % max_cpus;
+	    counter++;
+	} while(((env->arconnect.idu_data[cirq].dest >> core_id) & 0x1) == 0);
+	qemu_irq_raise(get_cpu_for_core(core_id)->env.irq[EXT_IRQ + cirq]);
+	return;
+	break;
+    case FIRST_ACKNOWLEDGE:
```

Comment:
IIRC this mode is not used in either Zephyr RTOS or the Linux kernel, so it's not clear how we're going to test it. That's why maybe just insert a runtime assert here?